init.rb

```diff
@@ -52,93 +52,58 @@
-        "distributions (Ubuntu 16.04 and later). Use 'stable' only for ",
-        "now officially unsupported distributions (Ubuntu 14.04)"])
-
-if Rock.in_release? && !Autoproj.config.has_value_for?('ROCK_SELECTED_FLAVOR')
-    Autoproj.config.set 'ROCK_SELECTED_FLAVOR', 'stable', true
-end
-
-Rock.flavors.select_current_flavor_by_name(
-    ENV['ROCK_FORCE_FLAVOR'] || Autoproj.config.get('ROCK_SELECTED_FLAVOR'))
-
-current_flavor = Rock.flavors.current_flavor
-
-#This check is needed because the overrides file will override the FLAVOR selection.
-#Furthermore a selection != stable can cause a inconsistent layout (cause by in_flavor system in the package_sets)
-if Rock.in_release? && current_flavor.branch != "stable" 
-    if ENV['ROCK_RC'] == '1'
-        Autoproj.warn ""
-        Autoproj.warn "Found a release file and the flavor is not master"
-        Autoproj.warn "This would usually be an error, but since ROCK_RC is set to 1,"
-        Autoproj.warn "it is assumed that you're preparing a release candidate"
-        Autoproj.warn ""
-    else
-        Autoproj.error ""
-        Autoproj.error "You selected the flavor '#{current_flavor.branch}' but '#{File.join(Autoproj.root_dir,"autoproj", "overrides.d", "25-release.yml")}' exists."
-        Autoproj.error "This means you are on a release; either unselect the release by calling 'rock-release switch master'"
-        Autoproj.error "or call 'autoproj reconfigure' and select the FLAVOR 'stable'"
-        exit 1
-    end
-end
 
-Autoproj.config.set('ROCK_SELECTED_FLAVOR', current_flavor.name, true)
-Autoproj.config.set('ROCK_FLAVOR', current_flavor.branch, true)
-Autoproj.config.set('ROCK_BRANCH', current_flavor.branch, true)
-
-if current_flavor.name != 'master' && Autoproj::PackageSet.respond_to?(:add_source_file)
-    Autoproj::PackageSet.add_source_file "source-stable.yml"
+# backward compatibility for the deprecated flavor system
+if !Autoproj.config.has_value_for?('ROCK_SELECTED_FLAVOR') then
+    Autoproj.config.set('ROCK_SELECTED_FLAVOR', "master")
 end
-
-require File.join(__dir__, 'rock', 'cxx')
-if Autoproj.respond_to?(:workspace) # autoproj 2.0
-    Rock.setup_cxx_support(Autoproj.workspace.os_package_resolver, Autoproj.config)
-else
-    Rock.setup_cxx_support(Autoproj.osdeps, Autoproj.config)
+if !Autoproj.config.has_value_for?('ROCK_FLAVOR') then
+    Autoproj.config.set('ROCK_FLAVOR', "master")
+end
+if !Autoproj.config.has_value_for?('ROCK_BRANCH') then
+    Autoproj.config.set('ROCK_BRANCH', "master")
 end
 
+
 def enabled_flavor_system
-    Rock.flavors.register_flavored_package_set(Autoproj.current_package_set)
+    Autoproj.warn "Flavors system was removed, please remove flavor-related code from your package_sets"
 end
 
 def in_flavor(*flavors, &block)
-    Rock.flavors.in_flavor(*flavors, &block)
+    Autoproj.warn "Flavors system was removed, please remove flavor-related code from your package_sets"
+    return false
```

Comment:
Need to yield for it to be a deprecation. The packages are defined in the block.